src/Microsoft.TemplateEngine.Edge/Template/TemplateCreator.cs

```diff
@@ -105,10 +105,36 @@ public TemplateCreator(IEngineEnvironmentSettings environmentSettings)
 
             string? realName = name is not null ? name : templateInfo.PreferDefaultName ? templateInfo.DefaultName : null;
             realName = realName is null ? fallbackName : realName;
+            if (!string.IsNullOrEmpty(name))
+            {
+                realName = name;
+            }
+            else if (templateInfo.PreferDefaultName)
+            {
+                if (string.IsNullOrEmpty(templateInfo.DefaultName))
+                {
+                    return new TemplateCreationResult(CreationResultStatus.TemplateIssueDetected, template.Name, "--name");
```

Comment:
`TemplateCreationResult` should contain the localized error message specifying the issue and call to action to resolve it.

In case of `dotnet new` this message will be printed out here: https://github.com/dotnet/sdk/blob/0fdccd38a185af6d8f9fa74d7d386ebdef2c58d7/src/Cli/Microsoft.TemplateEngine.Cli/TemplateInvoker.cs#L283-L288

Now it looks as only `--name` will be printed.

Could you please check this?